moshi/src/main/java/com/squareup/moshi/ClassJsonAdapter.java

```diff
@@ -95,7 +95,13 @@ private void createFieldBindings(
         // Look up a type adapter for this type.
         Type fieldType = resolve(type, rawType, field.getGenericType());
         Set<? extends Annotation> annotations = Util.jsonAnnotations(field);
-        JsonAdapter<Object> adapter = moshi.adapter(fieldType, annotations);
+        JsonAdapter<Object> adapter;
+        try {
+          adapter = moshi.adapter(fieldType, annotations);
+        } catch (IllegalArgumentException e) {
```

Comment:
This is a bit specific. Worth doing `RuntimeException` since we're running user code here?

Reply:
was mirroring https://github.com/square/moshi/blob/29bdc0aa45cc80569c2524b2f8e370d0649bdd80/moshi/src/main/java/com/squareup/moshi/AdapterMethodsFactory.java#L53

i think you're right.